cromwell_tools/_cromwell_auth.py

```diff
@@ -0,0 +1,191 @@
+            - OAuth2 through Google by passing a service account JSON key file.
+            - HTTPBasic Authentication.
+            - No Auth
+
+        Args:
+            url (str): The URL to the Cromwell server. e.g. "https://cromwell.server.org/"
+            header (dict or None): Dictionary containing the (bearer token) authorization header.
+            auth (requests.auth.HTTPBasicAuth or None): HTTP Basic Authentication information,
+                i.e. username and password.
+        """
+
+        self._validate_auth(header, auth)
+        self.header = header
+        self.auth = auth
+
+        self.url = self._validate_url(url)
+
+    @staticmethod
+    def _validate_auth(header, auth):
+        """Validate the format of the auth header and the HTTPBasic Auth credentials.
+
+        Args:
+            header (dict or None): Dictionary containing the (bearer token) authorization header.
+            auth (requests.auth.HTTPBasicAuth or None): HTTP Basic Authentication information,
+                i.e. username and password.
+
+        Raises:
+            TypeError: when no credentials are provided or auth is not a valid HTTPBasicAuth object.
+            ValueError: when the header is not a valid header(with Bearer token).
+        """
+        if not header and not auth:
+            raise ValueError("Either a header containing bearer token or a HTTPBasic Auth object must be passed.")
+
+        if header is not None:
+            if not isinstance(header, dict):
+                raise TypeError('If passed, header must be a dict.')
+            if "Authorization" not in header.keys():
+                raise TypeError('The header must have an "Authorization" key')
+
+        if auth is not None:
+            if not isinstance(auth, requests.auth.HTTPBasicAuth):
+                raise TypeError('The auth object must be a valid "requests.auth.HTTPBasicAuth" object!')
+
+    @staticmethod
+    def _validate_url(url):
+        """Validate the input Cromwell url.
+
+        Args:
+            url (str): The URL to the Cromwell server. e.g. "https://cromwell.server.org/"
+
+        Returns:
+            str: The URL to the Cromwell server without slash at the end. e.g. "https://cromwell.server.org"
+
+        Raises:
+            ValueError: If the input url is invalid, i.e. not following http(s) schema.
+        """
+        if isinstance(url, str) and url.startswith('http'):
+            return url.strip('/')
+        else:
+            raise ValueError("url must be an str that points to an http(s) endpoint.")
+
+    @staticmethod
+    def _validate_connection(header, auth, url):
+        warnings.warn("This function doesn't work for Cromwell until Cromwell adds the endpoint for validating users'"
+                      "Auth information, e.g. a Who am I endpoint.", DeprecationWarning)
+        return None
+
+        dummy_auth = namedtuple('dummy_auth', ['header', 'auth', 'url'])
```

Comment:
Good point, I'm taking it out for now! It was there in the stale PR but I realized that it can't work, since the `/health` endpoint of Cromwell it's using now doesn't require auth at all. Returning `None` is like a short-out workaround, but I agree that taking it out and make a ticket is a better way!!